tests/ConditionalExecution.php

```diff
@@ -0,0 +1,94 @@
+<?php
+namespace Psalm\Tests;
+
+use UnexpectedValueException;
+use function preg_quote;
+use Psalm\Config;
+use Psalm\Context;
+use Psalm\Internal\RuntimeCaches;
+use Psalm\Tests\Internal\Provider;
+use function strpos;
+use function substr;
+
+class ConditionalExecution extends TestCase
+{
+    /** @var \Psalm\Internal\Analyzer\ProjectAnalyzer */
+    protected $project_analyzer;
+
+    public function setUp() : void
+    {
+        RuntimeCaches::clearAll();
+
+        $this->file_provider = new Provider\FakeFileProvider();
+
+        $this->project_analyzer = new \Psalm\Internal\Analyzer\ProjectAnalyzer(
+            new TestConfig(),
+            new \Psalm\Internal\Provider\Providers(
+                $this->file_provider,
+                new Provider\FakeParserCacheProvider()
+            )
+        );
+    }
+
+    /**
+     * @dataProvider providerValidCodeParse
+     *
+     * @param string $code
+     * @param array<string> $error_levels
+     */
+    public function testValidCode($code, array $error_levels = []): void
+    {
+        $test_name = $this->getTestName();
+        if (strpos($test_name, 'SKIPPED-') !== false) {
+            $this->markTestSkipped('Skipped due to a bug.');
+        }
+
+        $version_position = strpos($test_name, 'PHP-');
+        if ($version_position === false) {
+            throw new UnexpectedValueException('No PHPVersion in test name');
+        }
+
+        $version = substr($test_name, $version_position + 4, 3);
+        $this->project_analyzer->setPhpVersion($version);
+
+        $file_path = self::$src_dir_path . 'somefile.php';
+
+        $this->addFile(
+            $file_path,
+            $code
+        );
+
+        foreach ($error_levels as $error_level) {
+            $this->project_analyzer->getCodebase()->config->setCustomErrorLevel($error_level, Config::REPORT_SUPPRESS);
+        }
+
+        $this->analyzeFile($file_path, new Context());
+    }
+
+    /**
+     * @return array<string, array{string,error_levels?:string[]}>
+     */
+    public function providerValidCodeParse(): array
+    {
+        return [
+            'PHP-7.2-hrtime' => [
+                '<?php
+                    $_startTime = \PHP_VERSION_ID >= 70300 ? \hrtime(false)[0] : \time();'
+            ],
+            'PHP-7.4-DateTimeCreateFromInterface' => [
+                '<?php
+                    namespace FooBar;
+
+                    class Datetime extends \DateTime
+                    {
+                        public static function createFromInterface(\DatetimeInterface $datetime): \DateTime
+                        {
+                            if (\PHP_VERSION_ID >= 80000) {
+                                return parent::createFromInterface($datetime);
+                            }
+                        }
+                    }'
```

Comment:
I'm surprised this doesn't fail on 7.4 because it should: https://3v4l.org/8mpLv#v730

Reply:
Mmh you make me second guess myself! It's the whole point of this PR, it doesn't fail in PHP 7.4 because the body of the if is behind a PHP_VERSION check, no?

Reply:
oh, I see your point, there is no return anymore in this function

Reply:
I removed the return type that wasn't here in the original code (I added to to remove an unrelated Psalm issue). However, you're right, in normal conditions, Psalm throw a InvalidReturnType : `Not all code paths of FooBar\Datetime::createFromInterface end in a return statement, return type DateTime expected`

Reply:
Yeah, and I wonder why it did not do that here.

Reply:
I'm not sure I understood exactly what your modification does. Can you explain quickly?

Reply:
My understanding of that code is that returning `false` from `ExpressionResolver::enterConditional()` makes Psalm consider the `if` body as 'uninteresting', and then it doesn't do some of the things it does while walking the AST during scanning. The second part of the changes was to make Psalm omit scan of included files when in 'uninteresting' code.

Looking at the code, though, it doesn't seem like that flag affects method calls in any way. Does the test you added fail on master, i.e. does PR actually changes anything for method calls?